frontend_v2/src/app/components/publiclists/teamlist/teamlist.component.ts

```diff
@@ -633,4 +642,32 @@ export class TeamlistComponent implements OnInit, OnDestroy {
     return deleteTeamMember;
   }
 
+  /**
+  * Filter teams by team
+  * @param TeamName Participant team name
+  */
+  filterTeam(teamName) {
+    const SELF = this;
+    SELF.filterTeamsQuery = teamName;
+    let API_PATH = SELF.endpointsService.FilteredParticipantTeamURL(teamName);
+    if (SELF.isHost) {
+      API_PATH = SELF.endpointsService.FilteredHostTeamURL(teamName);
+    }
+    SELF.apiService.getUrl(API_PATH).subscribe(
+      data => {
+        console.log('Data', data);
+        if (data['results']) {
+          SELF.allTeams = data['results'];
+          if (SELF.isHost || SELF.isOnChallengePage) {
+            SELF.allTeams = SELF.appendIsSelected(SELF.allTeams);
+          }
+          SELF.updateTeamsView(true);
```

Comment:
I think this part is same for `fetchTeams` function so can we make it common and use it from there in both the places.